File: Beecrowd/1179 Preenchimento de Vetor IV/Main.java
import java.io.IOException;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);

        final int TAM = 5;
        final int NUM_INT = 15;
        int[] vP = new int[TAM];
        int[] vI = new int[TAM];
        int P = 0, I = 0, x;

        for (int i = 0; i < NUM_INT; i++) {
            x = sc.nextInt();

            if (x % 2 == 0) {
                vP[P] = x;
                P++;
                if (P == TAM) {
                    for (int j = 0; j < TAM; j++)
                        System.out.println("par[" + j + "] = " + vP[j]);
                    P = 0;
                }
            } else {
                vI[I] = x;
                I++;
                if (I == TAM) {
                    for (int j = 0; j < TAM; j++)
                        System.out.println("impar[" + j + "] = " + vI[j]);
                    I = 0;
                }
            }
        }

        for (int j = 0; j < I; j++)
            System.out.println("impar[" + j + "] = " + vI[j]);
        I = 0;

        for (int j = 0; j < P; j++)
            System.out.println("par[" + j + "] = " + vP[j]);
        P = 0;

        sc.close();
    }

}

File: Beecrowd/1179 Preenchimento de Vetor IV/main.cpp
#include <iostream>

using namespace std;

int main()
{

    int Vp[5], Vi[5], P = 0, I = 0, j, i, X;

    for (i = 0; i < 15; i++)
    {
        cin >> X;

        if (X % 2 == 0)
        {
            Vp[P] = X;
            P++;
            if (P == 5)
            {
                for (j = 0; j < 5; j++)
                    cout << "par[" << j << "] = " << Vp[j] << endl;
                P = 0;
            }
        }
        else
        {
            Vi[I] = X;
            I++;
            if (I == 5)
            {
                for (j = 0; j < 5; j++)
                    cout << "impar[" << j << "] = " << Vi[j] << endl;
                I = 0;
            }
        }
    }
    for (j = 0; j < I; j++)
        cout << "impar[" << j << "] = " << Vi[j] << endl;
    I = 0;

    for (j = 0; j < P; j++)
        cout << "par[" << j << "] = " << Vp[j] << endl;
    P = 0;

    return 0;
}